### PR TITLE
[cli] Use `output.stopSpinner()` interface consistently

### PR DESCRIPTION
**Pull request by TooTallNate**
There was an issue with `output.spinner()` spinner if the old-style interface was used (invoking the return value to stop the spinner), which caused the internal spinner instance in `output` to be defined, but in a "stopped" state. Then later the `text` gets updated to a different value but the spinner is still stopped and nothing gets rendered.

The fix is to ensure that `output.stopSpinner()` is called rather than directly invoking the cancel function. In fact, the return value of `output.spinner()` has been removed completely.

### CODE REVIEW

**Review by AndyBitz — DISMISSED** [review]

**Review by AndyBitz — DISMISSED** [review]

File: packages/now-cli/src/util/output/create-output.ts [hunk 1]
```diff
@@ -111,31 +112,29 @@ function _createOutput({ debug: debugEnabled = false }: OutputOptions = {}) {
...
     if (spinner) {
       spinner.text = message;
     } else {
       spinner = wait(message, delay);
     }
-    return spinner;
   }
 
   function stopSpinner() {
+    if (debugEnabled && spinnerMessage) {
```
Comment (styfle) [inline packages/now-cli/src/util/output/create-output.ts:129]: Is there a case where `spinnerMessage` has a value but `spinner` does not?

I would think this function could be simplified to something like:

```js
function stopSpinner() {
    if (spinner) {
      spinner();
      spinner = null;
      debug(`Spinner stopped (${spinnerMessage})`);
      spinnerMessage = '';
    }
}
```

File: packages/now-cli/src/util/output/create-output.ts [hunk 2]
```diff
@@ -22,14 +22,15 @@ export default function createOutput(opts?: OutputOptions) {
...
+  let spinnerMessage = '';
   let spinner: StopSpinner | null = null;
 
   function isDebugEnabled() {
     return debugEnabled;
   }
 
   function print(str: string) {
-    stopSpinner();
+    if (spinnerMessage) stopSpinner();
```
Comment (styfle) [inline packages/now-cli/src/util/output/create-output.ts]: Why conditionally call `stopSpinner()`? Seems like you could handle that in the `stopSpinner()` function.

File: packages/now-cli/src/util/output/create-output.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/now-cli/src/util/output/create-output.ts:129]: It's cause when debug is enabled, then `spinner` is never set.

File: packages/now-cli/src/util/output/create-output.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/now-cli/src/util/output/create-output.ts]: This one probably isn't necessary anymore.

**Review by styfle — APPROVED** [review]

File: packages/now-cli/src/util/output/create-output.ts (hunk 1 above)
Comment (styfle) [reply packages/now-cli/src/util/output/create-output.ts:129]: How about this:

```js
function stopSpinner() {
    debug(`Spinner stopped (${spinnerMessage})`);
    spinnerMessage = '';
    if (spinner) {
      spinner();
      spinner = null;
    }
}
```

**Review by AndyBitz — APPROVED** [review]

**Pull request merged**